packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -198,16 +201,19 @@ export class KubernetesClient {
     this.kubeConfigWatcher.onDidChange(async () => {
       this._onDidUpdateKubeconfig.fire({ type: 'UPDATE', location });
       await this.refresh();
+      this.apiSender.send('kubernetes-context-update');
     });
 
     this.kubeConfigWatcher.onDidCreate(async () => {
       this._onDidUpdateKubeconfig.fire({ type: 'CREATE', location });
       await this.refresh();
+      this.apiSender.send('kubernetes-context-create');
     });
 
     this.kubeConfigWatcher.onDidDelete(() => {
       this._onDidUpdateKubeconfig.fire({ type: 'DELETE', location });
       this.kubeConfig = new KubeConfig();
+      this.apiSender.send('kubernetes-context-delete');
```

Comment:
so it seems we're never looking at these events so maybe we should not send them instead ?

or you send kubernetes-context-event with the type 'DELETE', 'UPDATE' or 'CREATE' by being a client of onDidUpdateKubeconfig